src/utils/types.ts

```diff
@@ -291,16 +296,19 @@ type PageElements = {
 
 interface PageNEW {
   number: number
+  sectionCode: string
   name: string
   progress: Progress
+  reviewProgress?: ReviewProgress
   state: PageElement[]
 }
 
 type PageElement = {
-  element: ElementBaseNEW | ElementStateNEW
+  element: ElementStateNEW
   response: ResponseFull | null
   thisReviewLatestResponse?: ReviewResponse
   review?: ReviewQuestionDecision
+  isAssigned?: boolean
```

Comment:
I am thinking, if we had multiple types of structure with non optionals for all of the use permutations of structures, it would be super difficult to keep track of what structure is used where. I think it easier for now to just add optionals ?

Reply:
Oh! I see because this could be an Application structure - when seeing by a Applicant user.

What about using the `review?` to keep all review related fields?

Reply:
> Oh! I see because this could be an Application structure - when seeing by a Applicant user.

I think my concern is, we dont' set all of the fields at once on the structure, so that we would need to keep on changing the type of structure as we add fields ? (or make them optional)

And review field would have some missing fields within it too ?

One way to solve this is to have a fully constructed default structure at the start. But I think optionals is fine for now